lib/model/menu/menu.dart

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:meta/meta.dart';
+import 'package:observable/observable.dart';
+import 'package:quiver/strings.dart' show isNotEmpty;
+import 'package:angular_components/model/a11y/active_item.dart';
+import 'package:angular_components/model/collection/combined_list.dart';
+import 'package:angular_components/model/collection/labeled_list.dart';
+import 'package:angular_components/model/observable/observable.dart';
+import 'package:angular_components/model/ui/display_name.dart';
+import 'package:angular_components/model/ui/icon.dart';
+import 'package:angular_components/utils/id_generator/id_generator.dart';
+
+import 'menu_item_affix.dart';
+
+export 'menu_item_affix.dart';
+
+/// A list of menu items with an optional name.
+class MenuItemGroup<T> extends LabeledList<T> {
```

Comment:
I'm not sure about the motive but MenuModel is described as being immutable in the comment. AcceptsWidth might have been left off to avoid encouraging someone to try and change the width.

Reply:
Oh, right! Then in line 63
```dart
int width;
```

should probably be `final`, right?